src/client/cl_entity.c

```diff
@@ -87,23 +96,8 @@ static void Cl_ReadDeltaEntity(cl_frame_t *frame, entity_state_t *from, uint16_t
 	ent->frame_num = cl.frame.frame_num;
 	ent->current = *to;
 
-	// update clipping matrices, snapping the entity to the frame
-	if (ent->current.solid) {
-		if (ent->current.solid == SOLID_BSP) {
-			if (bits & (U_ORIGIN | U_ANGLES)) {
-				Matrix4x4_CreateFromEntity(&ent->matrix, ent->current.origin, ent->current.angles, 1.0);
-				Matrix4x4_Invert_Simple(&ent->inverse_matrix, &ent->matrix);
-			}
-		} else { // bounding-box entities
-			if (bits & U_ORIGIN) {
-				Matrix4x4_CreateFromEntity(&ent->matrix, ent->current.origin, vec3_origin, 1.0);
-				Matrix4x4_Invert_Simple(&ent->inverse_matrix, &ent->matrix);
-			}
-		}
-	}
-
```

Comment:
Rather than only updating these matrices when the entity is updated here, I've moved this code into `Cg_Interpolate` so that (when we resolve the prediction issues), we can actually interpolate the collision matrix, and have pixel-perfect client-side collision.